### DISCUSSION TITLE
Forecast AutoARIMA with Exogenous variables

### DISCUSSION THREAD

**Discussion by AFilipaVEL** [Nixtla/statsforecast#869] [General]
Hi all! 

I'm trying to train an AutoARIMA model with an exogenous variable (which has no meaning, i'm just trying to see if the code runs), in a Spark environment. 

When i forecast without the variable, everything runs ok but as i include the X_df dataframe i get the error

**'ValueError: The time column ('ds') should have either timestamps or integers, got 'object'.'**

My code is:

```
models = [AutoARIMA()]

sf = StatsForecast(models=models,  freq='M', n_jobs=1)

fcst = sf.forecast(df= train, h=1, X_df=X_test, level = [90])

fcst.head() 
```
The train and X_test dataframes are dataframes like the ones in the example of https://nixtlaverse.nixtla.io/statsforecast/docs/how-to-guides/exogenous.html

Do you have any idea of what is going on? 

Thanks in advance! 
Filipa

**Comment by jmoralez** [comment]
The fix was casting the time column to timestamp: `df.withColumn('ds', col('ds').cast(TimestampType()))`
